### PR TITLE
Misc map fixes

### PR DESCRIPTION
**Pull request by num0005**
Fixes recent issues with custom maps not working.

### CODE REVIEW

File: xlive/H2Tweaks.cpp [hunk 1]
```diff
@@ -515,13 +528,28 @@ int __cdecl validate_and_add_custom_map(BYTE *a1)
 	if (!validate_and_add_custom_map_interal_impl(a1))
 	{
 		TRACE_FUNC("warning \"%s\" has bad checksums or is blacklisted, map may not work correctly", file_name);
+		std::wstring fallback_name;
+		if (strnlen_s(header.name, sizeof(header.name)) > 0)
```
Comment (shayaantx) [inline xlive/H2Tweaks.cpp]: open brace on same line

Comment (shayaantx) [reply xlive/H2Tweaks.cpp]: or make else open brace on next line

File: xlive/Util/filesys.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include <stdafx.h>
+#include "filesys.h"
+
+std::wstring GetExeDirectoryWide()
+{
+	wchar_t main_exe_dir[MAX_PATH];
+	GetModuleFileNameW(NULL, main_exe_dir, sizeof(main_exe_dir));
+	size_t path_len = wcsnlen_s(main_exe_dir, sizeof(main_exe_dir));
```
Comment (shayaantx) [inline xlive/Util/filesys.cpp:9]: duplicated code between both these methods

**Review by shayaantx — CHANGES_REQUESTED** [review]

File: xlive/H2Tweaks.cpp (hunk 1 above)
Comment (num0005) [reply xlive/H2Tweaks.cpp]: one day I will bother to setup msvc correctly

**Review by shayaantx — APPROVED** [review]

**Pull request merged**